pkg/kubelet/eviction/helpers.go

```diff
@@ -562,12 +570,21 @@ func memory(stats statsFunc) cmpFunc {
 		}
 
 		// adjust p1, p2 usage relative to the request (if any)
+		var p1Request, p2Request resource.Quantity
 		p1Memory := memoryUsage(p1Stats.Memory)
-		p1Request := v1resource.GetResourceRequestQuantity(p1, v1.ResourceMemory)
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			p1Request = v1resource.GetResourceAllocationQuantity(p1, v1.ResourceMemory)
```

Comment:
I'd suggest checking the feature gate inside of this function, and returning GetResourceRequestQuantity if it isn't set, rather than having  all of these if/else statements.

Reply:
That's what I did at first, but when I read the code, it seemed to be hiding what's actually being returned. In particular, switching inside PodRequestsAndLimits worked fine for kubelet but changed the values seen by kubectl describe. We needed PodResourceAllocations, and after adding it, I felt that being explicit made it easier to read and understand what's going on. 

I'm worried that if we hide it inside, then down the road someone might use the function thinking they are getting Requests, while they are actually getting something else - and regress when we beta/GA this feature.

Once we GA and get rid of feature-gate, this will look clean.